hbase_sql/src/main/java/com/splicemachine/compactions/SpliceDefaultCompactor.java

```diff
@@ -115,14 +116,16 @@ public SpliceDefaultCompactor(final Configuration conf, final Store store, long
     @SuppressFBWarnings(value="ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD", justification="static attribute hostname is set from here")
     public List<Path> compact(CompactionRequestImpl request, ThroughputController throughputController, User user) throws IOException {
         assert request instanceof SpliceCompactionRequest;
+        SpliceCompactionRequest spliceRequest = (SpliceCompactionRequest)request;
         // Used if we cannot compact in Spark
-        ((SpliceCompactionRequest) request).setPurgeConfig(buildPurgeConfig(request));
+        spliceRequest.setPurgeConfig(buildPurgeConfig(request));
 
-        if(!allowSpark || store.getRegionInfo().getTable().isSystemTable())
+        if(!allowSpark || store.getRegionInfo().getTable().isSystemTable()) {
+            isSpark = false;
```

Comment:
Setting isSpark here is redundant

Reply:
Right, but I would still leave it there to protected against e.g. accidentally initializing the member variable to `true`